### DISCUSSION TITLE
grpc.Client.waitForReady?

### DISCUSSION THREAD

**Discussion by swwilson-bsi** [timostamm/protobuf-ts#176] [Q&A]
How can a client wait for a server to be ready? After creating a GrpcTransport, I want to call client.waitForReady. However, client is a private property.

**Comment by timostamm** [comment]
It can't, at the moment. Could you elaborate why you want to call `waitForReady`?  
Not questioning your motives - I'd like to learn in which cases this would be beneficial.

**Reply by swwilson-bsi** [reply]
I found that the first rpc method call would time out if the server wasn't ready (still starting up). I tried setting the timeout property of the transport object, but it didn't seem to help. BTW In my case, both client and server are processes on the same machine. The client launches the server.

**Reply by timostamm** [reply]
Interesting. I've been using retry policies for that purpose (that was not with gRPC though). Being able to wait for a connection independently of calls is nice to have, though. 

I think it would be reasonable to expose `waitForReady` on the transport, similar to what we already do with `close`:

https://github.com/timostamm/protobuf-ts/blob/376f56e91e0558505ba4f6702fe1d381b53862de/packages/grpc-transport/src/grpc-transport.ts#L311-L313

Would you be interested in opening a PR for this feature?

**Comment by swwilson-bsi** [comment]
Gee. I wouldn't know how to test it. Would you do it?

**Reply by timostamm** [reply]
Yes. I'll review the PR and publish a pre-release version.
